Marlin/src/module/stepper.cpp

```diff
@@ -521,7 +521,7 @@ void Stepper::enable_axis(const AxisEnum axis) {
 bool Stepper::disable_axis(const AxisEnum axis) {
   mark_axis_disabled(axis);
 
-  TERN_(DWIN_LCD_PROUI, set_axis_untrusted(axis)); // MRISCOC workaround: https://github.com/MarlinFirmware/Marlin/issues/23095
+  TERN_(HOME_AFTER_DEACTIVATE, set_axis_untrusted(axis)); // MRISCOC workaround: https://github.com/MarlinFirmware/Marlin/issues/23095
```

Comment:
Axes are always marked as "untrusted" when they are disabled (below) by `DISABLE_AXIS_##N()`. The workaround added here for the benefit or ProUI is unclear. The `can_axis_disable` used below is meant to check whether a stepper may be in use by more than one axis, as with the case of CoreXZ, for example. This workaround may be intended to force behavior that would not otherwise apply in a shared stepper situation. Maybe it can be removed.

Reply:
This looks like an attempt to ensure the axis get marked as untrusted as the disable function is not always correctly setting the untrusted flag.

Reply:
Yeah, this should really be removed then.

Reply:
This has been set back as it was, as it's not related to this PR.